src/Sylius/Bundle/CoreBundle/Checkout/Step/InventoryStep.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Checkout\Step;
+
+use Sylius\Bundle\FlowBundle\Process\Context\ProcessContextInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Sylius\Component\Core\SyliusCheckoutEvents;
+use Sylius\Component\Inventory\Manager\InsufficientRequirementsException;
+
+/**
+ * Inventory step.
+ *
+ * Checks to make sure all inventory is avilable
+ *
+ * @author Myke Hines <myke@webhines.com>
+ */
+class InventoryStep extends CheckoutStep
```

Comment:
I don't think this is good idea. IIRC we use CartResolver to check such things. I.e. have a look at https://github.com/Sylius/Sylius/pull/1935 that was one of ideas how to improve this stuff.